kivy/core/audio/audio_sdl2.pyx

```diff
@@ -149,7 +148,8 @@ class SoundSDL2(Sound):
         self.unload()
         if self.filename is None:
             return
-        mc.chunk = Mix_LoadWAV(<char *><bytes>self.filename)
+        fn = self.filename.encode('UTF-8')
```

Comment:
I'm guessing you tested this in py3? If `self.filename` is already `bytes()` then this will throw an exception. You should probably do something like `fn = self.filename.encode('utf8') if not isinstance(self.filename, bytes) else self.filename` instead.